cg/apps/cgstats/demux_sample.py

```diff
@@ -0,0 +1,151 @@
+    lane: int
+    raw_clusters_pc: float
+    pass_filter_clusters: int
+    pass_filter_yield_pc: float
+    pass_filter_yield: int
+    pass_filter_Q30: float
+    pass_filter_read1_q30: float
+    pass_filter_read2_q30: float
+    pass_filter_qscore: float
+    undetermined_pc: float
+    barcodes: int
+    perfect_barcodes: int
+    one_mismatch_barcodes: int = 0
+
+
+def create_demux_sample(
+    sample_id: str,
+    flowcell: str,
+    lane: int,
+    barcode_stats: SampleBarcodeStats,
+    conversion_stats: SampleConversionResults,
+    nr_raw_clusters: int,
+) -> DemuxSample:
+    return DemuxSample(
+        sample_name=sample_id,
+        flowcell=flowcell,
+        lane=lane,
+        raw_clusters_pc=round(conversion_stats.raw_cluster_count / nr_raw_clusters * 100, 2),
+        pass_filter_clusters=conversion_stats.pass_filter_cluster_count,
+        pass_filter_yield_pc=round(
+            conversion_stats.pass_filter_yield / conversion_stats.raw_yield * 100,
+            2,
+        )
+        if conversion_stats.raw_yield
+        else 0,
+        pass_filter_yield=conversion_stats.pass_filter_yield,
+        pass_filter_Q30=round(
+            conversion_stats.pass_filter_q30 / conversion_stats.pass_filter_yield * 100,
+            2,
+        )
+        if conversion_stats.pass_filter_yield
+        else 0,
+        pass_filter_read1_q30=round(
+            conversion_stats.pass_filter_read1_q30 / conversion_stats.pass_filter_read1_yield * 100,
+            2,
+        )
+        if conversion_stats.pass_filter_read1_yield
+        else 0,
+        pass_filter_read2_q30=round(
+            conversion_stats.pass_filter_read2_q30 / conversion_stats.pass_filter_read2_yield * 100,
+            2,
+        )
+        if conversion_stats.pass_filter_read2_yield
+        else 0,
+        pass_filter_qscore=round(
+            conversion_stats.pass_filter_quality_score_sum / conversion_stats.pass_filter_yield,
+            2,
+        )
+        if conversion_stats.pass_filter_yield
+        else 0,
+        undetermined_pc=(conversion_stats.pass_filter_cluster_count - barcode_stats.barcode_count)
+        / conversion_stats.pass_filter_cluster_count
+        * 100
+        if conversion_stats.pass_filter_cluster_count
+        else 0,
+        barcodes=barcode_stats.barcode_count,
+        perfect_barcodes=barcode_stats.perfect_barcode_count,
+        one_mismatch_barcodes=barcode_stats.one_mismatch_barcode_count,
+    )
+
+
+def get_demux_samples(
+    conversion_stats_path: Path, demux_stats_path: Path, sample_sheet: SampleSheet
+) -> Dict[int, Dict[str, DemuxSample]]:
+    """Gather information from demultiplexing results and create samples with the correct information"""
+    demux_samples: Dict[int, Dict[str, DemuxSample]] = {}
+    demux_stats_parser: DemuxStats = DemuxStats(stats_file=demux_stats_path)
+    conversion_stats_parser: ConversionStats = ConversionStats(
+        conversion_stats=conversion_stats_path
+    )
```

Comment:
I think this is a bit confusing! 
variable is conversion_stats_parser, made from ConversionStats class, which takes as a "conversion_stats" argument a path to stats file. The there is a variable on ln 107 which is conversion_stats made from SampleConversionResults

I suggest harmonize names of objects with names of classes they are made from

Reply:
better now?